packages/matter.js/src/cluster/server/CommandServer.ts

```diff
@@ -27,8 +53,19 @@ export class CommandServer<RequestT, ResponseT> {
     async invoke(session: Session<MatterDevice>, args: TlvStream, message: Message, endpoint: Endpoint): Promise<{ code: StatusCode, responseId: number, response: TlvStream }> {
         const request = this.requestSchema.decodeTlv(args);
         logger.debug(`Invoke ${this.name} with data ${Logger.toJSON(request)}`);
-        const response = await this.handler(request, session, message, endpoint);
-        logger.debug(`Invoke ${this.name} response : ${Logger.toJSON(response)}`);
-        return { code: StatusCode.Success, responseId: this.responseId, response: this.responseSchema.encodeTlv(response) };
+        try {
```

Comment:
Not needed when using StatusResponseError because handling for this is already existing "upstream" in code

Reply:
I could be reading it wrong but it looks like the behavior will be different in handleInvokeResponse:

* If you throw it will go into the Promise.all which will then throw from handleInvokeRequest
* If you return a code then it will go into the invokeResponses array so other invocations could potentially succeed

But larger question is whether the approach is valid at all.  Last time we discussed I got the impression we needed to return the appropriate status code.  This solves for that, but it doesn't solve for any type of introspection.  Sounds like we need to be able to report on which commands are supported separately from invocation?

Reply:
The currentl code will catch that even one level higher.
I have one todo on my plate to make sure we answer all sorts of reponses correctly in all cases - like that case.
So this try catch that you added there I would move to https://github.com/project-chip/matter.js/blob/main/packages/matter.js/src/protocol/interaction/InteractionServer.ts#L431 here ... and then do generically ... when "StatusResponseError" then get the code out of the error and use this as response code and add reponse like that

```
                invokeResponses.push({ status: { commandPath, status: { status: code } } });
```

If you like do that already ... then I check the rest cases later.

The "not implemented" cases we have in code right now were throwing an "neutral exception" so that we see if someone is causing them because in fact it is more like "TODO needs to be implemented" instead of a valid "ok to not have them", thats whey they also till now were not really implemented as real errors.
So question is if we leave it that way or "clean it up" like in your approach.
Ideally they should all be implemented correctly by us (core logic) or "the dev".

Reply:
I'd probably make it explicit with IncompleteServerError extends StatusResponseError or somesuch but LMK what you'd prefer, I'll change the ones in the handlers to that.

Reply:
subclasses are fine, so we can still catch StatusResponseError generically ... thats what I would like to keep as option because a command can throw also other statusResponse errors nd so the catch is generic